### PR TITLE
Angular 2 Testing In Depth: Services

### PR DESCRIPTION
**Pull request by blacksonic**

### CODE REVIEW

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+  bg_merge: true
+tags:
+- angular 2
+- testing
+- angular
+- typescript
+- jasmine
+---
+
+When i started developing and writing tests for Angularjs applications,
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Capitalize "I" (everywhere).

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+everything felt natural. The tools were mature and i easily got used to
+develop applications in TDD (Test Driven Development).
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Grammar: "developing".

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+It gave me a high level of confidence, that my application is working as i imagined.
+Just after Angular 2 came out and i learnt the basics, i knew that the next step is to learn testing with it.
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: ...the next step *was* to...

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+where i share my experiences testing different building blocks of an Angular 2 application.
+We will start with simple use cases and then head for more complex ones.
+
+- Services (this article)
+- Http Services
+- Components
+- Pipes
+- Routing
+
+You may wonder why is it so important to write tests.
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: ...why it is...

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+
+With tests, we can guard against someone breaking our code by refactoring or
+adding new features. This might have happened to you when someone added a small feature
+or added equivalent code transformations, and nothing worked afterwards.
+Writing tests can clarify the intention of the code by giving usage examples.
+It can also reveal design flaws. When a piece of code is hard to test,
+there might be a problem with the underlying architecture.
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Consider remarking tests are also important to make sure the code does what it was designed to do. Consider putting a link to "test driven development" definitions.

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+
+### Choosing the framework
+
+The first thing we have to choose is the framework. The suggested one by Angular 2's core team is Jasmine.
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Consider putting a link to the `Jasmine` homepage.

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+For a long time it was the only supported testing framework, because test setups were hard wired into the framework.
+Thanks to refactoring now tests can also be written in Mocha, or any other framework
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Consider putting a link to the `Mocha` homepage.

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+it('should display name with engine', inject([Car], (car: Car) => {
+  expect(car.getName()).toEqual('Car with Basic engine(150 HP)');
+}));
+```
+
+### Mocking
+
+In unit tests we want to execute the code in isolation.
+It means not depending on big complex objects and not calling methods that rely on external systems
+(like Http calls or database access).
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: "Http" -> "HTTP"

File: _posts/2016-09-19-angular-2-testing-series-services.markdown [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+### Conclusion
+
+In this tutorial, we managed to:
+
+- create a basic service and write tests for it
+- use dependency injection in tests
+- fake dependencies with Jasmine
+- fake dependencies with dependency injection
+
+Hope it convinced you, that writing tests in Angular 2 is not an overly complicated thing.
```
Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Use no comma here.

File: _posts/2016-09-19-angular-2-testing-series-services.markdown (hunk 2 above)
Comment (blacksonic) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: @sebadoom Can you elaborate on this? Should i rephrase it to "...started to develop and write tests..."?

Comment (sebadoom) [inline _posts/2016-09-19-angular-2-testing-series-services.markdown]: Next sentence: ...I easily got used to *developing*...
